src/AppInstallerRepositoryCore/Rest/Schema/1_0/Interface.cpp

```diff
@@ -74,18 +73,47 @@ namespace AppInstaller::Repository::Rest::Schema::V1_0
             return OptimizedSearch(request);
         }
 
-        // TODO: Handle continuation token
-        std::optional<web::json::value> jsonObject = m_httpClientHelper.HandlePost(
-            m_searchEndpoint, GetSearchBody(request), m_requiredRestApiHeaders);
+        return SearchInternal(request);
+    }
 
-        if (!jsonObject)
+    IRestClient::SearchResult Interface::SearchInternal(const SearchRequest& request) const
+    {
+        SearchResult results;
+        utility::string_t continuationToken;
+        std::unordered_map<utility::string_t, utility::string_t> searchHeaders = m_requiredRestApiHeaders;
+        do
+        {
+            if (!continuationToken.empty())
+            {
+                AICLI_LOG(Repo, Verbose, << "Received continuation token. Retrieving more results.");
+                searchHeaders.insert_or_assign(JsonHelper::GetUtilityString(ContinuationToken), continuationToken);
+            }
+
+            std::optional<web::json::value> jsonObject = m_httpClientHelper.HandlePost(m_searchEndpoint, GetSearchBody(request), searchHeaders);
+
+            utility::string_t ct;
+            if (jsonObject)
+            {
+                SearchResponseDeserializer searchResponseDeserializer;
+                SearchResult currentResult = searchResponseDeserializer.Deserialize(jsonObject.value());
+                
+                size_t insertElements = !request.MaximumResults ? currentResult.Matches.size() :
+                    std::min(currentResult.Matches.size(), request.MaximumResults - results.Matches.size());
+
+                std::move(currentResult.Matches.begin(), std::next(currentResult.Matches.begin(), insertElements), std::inserter(results.Matches, results.Matches.end()));
+                ct = RestHelper::GetContinuationToken(jsonObject.value()).value_or(L"");
+            }
+
+            continuationToken = ct;
+
+        } while (!continuationToken.empty() && (!request.MaximumResults || results.Matches.size() < request.MaximumResults));

```

Comment:
This really suggests a need to refactor the interface to allow for an actual iteration of the results rather than a simple list containing all of them.  We need to make sure that the COM API is set up to handle that from the start.

@sreadingMSFT , should we change the `FindPackagesResult` to have an `IIterable` rather than an `IVectorView` for future consideration?

Reply:
Sounds reasonable. Let me know whenever this needs to be updated/refactored for COM Api's use.